File: sdk/c#/PicarioXPO.RenderAPI/XpoUrlTemplate.cs
namespace PicarioXPO.RenderAPI
{
    /// <summary>
    /// Denines a PicarIS template
    /// </summary>
    public sealed class XpoUrlTemplate
    {
        /// <summary>
        /// Gets or sets the index of this template
        /// </summary>
        public int Index { get; set; }

        /// <summary>
        /// The template parameter value
        /// </summary>
        public string Value { get; set; }
    }
}

File: sdk/js/PicarioXPO.RenderAPI.V2/UrlGenerators/XpoUrlTemplate.ts
class XpoUrlTemplate {
    index: number;
    value: string;

    constructor(index: number, value: string) {
        this.index = index;
        this.value = value;
    }

    /*
     * Gets or sets the index of this template
     */
    getIndex(): number { return this.index; }
    setIndex(val: number) { this.index = val; }
    
    /*
     * Gets or sets the value of this template
     */
    getValue(): string { return this.value; }
    setValue(val: string) { this.value = val; }
}